external/solr/src/main/java/org/apache/stormcrawler/solr/persistence/SolrSpout.java

```diff
@@ -156,10 +176,15 @@ else if (resetFetchDateAfterNSecs != -1) {
         LOG.debug("QUERY => {}", query);
 
         try {
+            LOG.trace("isInQuery set to true");
+            isInQuery.set(true);
+
             long startQuery = System.currentTimeMillis();
             QueryResponse response = connection.getClient().query(query);
             long endQuery = System.currentTimeMillis();
 
+            markQueryReceivedNow();
+
```

Comment:
I guess the `isInQuery.set(true)` and `markQueryReceivedNow()` are only relevant when the calls to Solr are asynchronous right? If so, we will have to review this for #621.

Reply:
It is used in the super class and has an impact on when _populateBuffer_ is called
see [AbstractQueryingSpout](https://github.com/apache/incubator-stormcrawler/blob/6ba510c91848dd3d8bce09dc65ac97194eea829c/core/src/main/java/org/apache/stormcrawler/persistence/AbstractQueryingSpout.java#L182)